src/main/angular/app/project-select/project-select.component.ts

```diff
@@ -0,0 +1,199 @@
+  public projectId = SortBy.PROJECT_ID;
+  public projectNumber = SortBy.PROJECT_NUMBER;
+  // #endregion
+
+  @Output()
+  public changeProjects = new EventEmitter<Project[]>();
+
+  constructor(private dataService: DataService) {
+    this.query = '';
+    this.projects = [];
+    this.activeProjects = new Set();
+
+    this.currentSortDirection = SortDirection.DESCENDING;
+    this.currentSortField = SortBy.IAM_BINDINGS;
+  }
+
+  /** Returns the color associated with the given project. */
+  getColor(project: Project): string {
+    if (this.activeProjects.has(project)) {
+      return project.color;
+    }
+    return '#b8b8b8';
+  }
+
+  /** Toggles the given projects presence on the graph. */
+  toggleProject(project: Project) {
+    if (this.activeProjects.has(project)) {
+      this.activeProjects.delete(project);
+    } else {
+      this.activeProjects.add(project);
+    }
+    this.changeProjects.emit(Array.from(this.activeProjects));
+  }
+
+  /** Returns the class placed on a sorting arrow, which decides whether it's grayed out or not. */
+  getSortClass(field: SortBy) {
+    if (field === this.currentSortField) {
+      return 'sort-active';
+    }
+    return 'sort-inactive';
+  }
+
+  /** Returns the status (up or down) of the animation associated with the given sort field. */
+  getAnimationStatus(field: SortBy): 'down' | 'up' {
+    switch (field) {
+      case SortBy.IAM_BINDINGS:
+        return this.sortRotated.iamBindings;
+      case SortBy.NAME:
+        return this.sortRotated.projectName;
+      case SortBy.PROJECT_ID:
+        return this.sortRotated.projectId;
+      case SortBy.PROJECT_NUMBER:
+        return this.sortRotated.projectNumber;
+    }
+  }
+
+  /** Swap the given animation from up to down or vice versa. */
+  swapAnimationProperty(field: SortBy) {
+    switch (field) {
+      case SortBy.IAM_BINDINGS:
+        this.sortRotated.iamBindings =
+          this.sortRotated.iamBindings === 'down' ? 'up' : 'down';
+        break;
+      case SortBy.NAME:
+        this.sortRotated.projectName =
+          this.sortRotated.projectName === 'down' ? 'up' : 'down';
+        break;
+      case SortBy.PROJECT_ID:
+        this.sortRotated.projectId =
+          this.sortRotated.projectId === 'down' ? 'up' : 'down';
+        break;
+      case SortBy.PROJECT_NUMBER:
+        this.sortRotated.projectNumber =
+          this.sortRotated.projectNumber === 'down' ? 'up' : 'down';
+        break;
+    }
+  }
+
+  /** Change the sort field/direction, adjusts styling and sorts projects. */
+  changeSort(field: SortBy) {
+    this.currentSortField = field;
+    if (this.getAnimationStatus(field) === 'down') {
+      this.currentSortDirection = SortDirection.ASCENDING;
+    } else {
+      this.currentSortDirection = SortDirection.DESCENDING;
+    }
+    // Animate the selected field
+    this.swapAnimationProperty(field);
+  }
+
+  /** Returns a sorted and filtered view of the projects. */
+  getProjects(): Project[] {
```

Comment:
I believe that this method gets called on every digest loop, so every user click, manipulation, etc. This could cause significant lag over time as this list gets constructed and reconstructed. 

In a follow on PR, could you try to refactor this? A TODO is sufficient for now